provider/machine_pool_resource.go

```diff
@@ -785,21 +785,49 @@ func (r *MachinePoolResource) Delete(ctx context.Context, request tfsdk.DeleteRe
 		MachinePool(state.ID.Value)
 	_, err := resource.Delete().SendContext(ctx)
 	if err != nil {
-		response.Diagnostics.AddError(
-			"Can't delete machine pool",
-			fmt.Sprintf(
-				"Can't delete machine pool with identifier '%s' for "+
-					"cluster '%s': %v",
-				state.ID.Value, state.Cluster.Value, err,
-			),
-		)
-		return
+		// We can't delete the pool, see if it's the last one:
+		numPools, err2 := r.countPools(ctx, state.Cluster.Value)
+		if numPools == 1 && err2 == nil {
+			// It's the last one, issue warning instead of error
+			response.Diagnostics.AddWarning(
+				"Can't delete machine pool",
+				fmt.Sprintf(
```

Comment:
Per @arendej, add: "ROSA Classic clusters must have at least one machine pool.", and change "Can't" -> "Cannot"